Libraries/MLXVLM/VLMModelFactory.swift

```diff
@@ -0,0 +1,228 @@
+        "paligemma": create(PaliGemmaConfiguration.self, PaliGemma.init),
+        "qwen2_vl": create(Qwen2VLConfiguration.self, Qwen2VL.init),
+    ]
+
+    /// Add a new model to the type registry.
+    public func registerModelType(
+        _ type: String, creator: @Sendable @escaping (URL) throws -> any LanguageModel
+    ) {
+        lock.withLock {
+            creators[type] = creator
+        }
+    }
+
+    /// Given a `modelType` and configuration file instantiate a new `LanguageModel`.
+    public func createModel(configuration: URL, modelType: String) throws -> any LanguageModel {
+        let creator = lock.withLock {
+            creators[modelType]
+        }
+        guard let creator else {
+            throw ModelFactoryError.unsupportedModelType(modelType)
+        }
+        return try creator(configuration)
+    }
+
+}
+
+public class ProcessorTypeRegistry: @unchecked Sendable {
+
+    // Note: using NSLock as we have very small (just dictionary get/set)
+    // critical sections and expect no contention.  this allows the methods
+    // to remain synchronous.
+    private let lock = NSLock()
+
+    private var creators:
+        [String: @Sendable (URL, any Tokenizer) throws -> any UserInputProcessor] = [
+            "PaliGemmaProcessor": create(
+                PaliGemmaProcessorConfiguration.self, PaligGemmaProcessor.init),
+            "Qwen2VLProcessor": create(
+                Qwen2VLProcessorConfiguration.self, Qwen2VLProcessor.init),
+        ]
+
+    /// Add a new model to the type registry.
+    public func registerProcessorType(
+        _ type: String,
+        creator: @Sendable @escaping (URL, any Tokenizer) throws -> any UserInputProcessor
+    ) {
+        lock.withLock {
+            creators[type] = creator
+        }
+    }
+
+    /// Given a `processorType` and configuration file instantiate a new `UserInputProcessor`.
+    public func createModel(configuration: URL, processorType: String, tokenizer: any Tokenizer)
+        throws -> any UserInputProcessor
+    {
+        let creator = lock.withLock {
+            creators[processorType]
+        }
+        guard let creator else {
+            throw ModelFactoryError.unsupportedProcessorType(processorType)
+        }
+        return try creator(configuration, tokenizer)
+    }
+
+}
+
+/// Registry of models and any overrides that go with them, e.g. prompt augmentation.
+/// If asked for an unknown configuration this will use the model/tokenizer as-is.
+///
+/// The python tokenizers have a very rich set of implementations and configuration.  The
+/// swift-tokenizers code handles a good chunk of that and this is a place to augment that
+/// implementation, if needed.
+public class ModelRegistry: @unchecked Sendable {
+
+    private let lock = NSLock()
+    private var registry = Dictionary(uniqueKeysWithValues: all().map { ($0.name, $0) })
+
+    static public let paligemma3bMix4488bit = ModelConfiguration(
```

Comment:
nit: but could we change the name to 
```suggestion
    static public let paligemma3bMix448_8bit = ModelConfiguration(
```

just so the quantization is a little more readable?